onepassword/onepassword.go

```diff
@@ -197,6 +197,24 @@ func (op *OnePassword) Get(ctx context.Context, secret Secret) (string, error) {
 	}
 }
 
+func (op *OnePassword) GetDocument(ctx context.Context, secret Secret) (string, error) {
+	if op.connect != nil {
+		if value := op.connectGetFileContent(secret.Field, secret.Vault, secret.Item); len(value) == 0 {
+			return "", fmt.Errorf("could not find document: '%s' '%s' '%s'", secret.Field, secret.Vault, secret.Item)
+		} else {
+			return value, nil
+		}
+	} else {
+		if ok, _ := op.Session(); !ok {
+			return "", ErrNotSignedIn
+		} else if value := op.clientGetDoument(ctx, secret.Vault, secret.Item); len(value) == 0 {
+			return "", fmt.Errorf("could not find document '%s' '%s'", secret.Vault, secret.Item)
+		} else {
+			return value, nil
+		}
```

Comment:
This is a really odd pattern. 
I've talked to many people about it, and not a single one preferred the chained else statement.
There is really nothing like this in the standard library.

Reply:
it's a more complex one to use scoped varialbes than what effective-go shows as example and use case:

"Since if and switch accept an initialization statement, it's common to see one used to set up a local variable."

if err := file.Chmod(0664); err != nil {
    log.Print(err)
    return err
}

Note: This example sucks cause it returns.

But I'll note that it's too complicated to understand and take that into further implementations.

Reply:
oh no, it's something else, this is the part of the nesting of if else calls! That's not the makefile!